docs/DeveloperGuide.md

```diff
@@ -5,9 +5,262 @@
 
 ## Design & implementation
+This section describes how certain features are implemented
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Input Parsing
+The program handles user inputs in the Parser class where inputs are parsed into command classes that implement the Command interface. 
+The Parser class only contains static methods as we have determined that it would be more practical instead of instantiating an instance of a Parser class.
 
+> Note on the command inputs:
+> - All valid command inputs by the user will have a command word _(first word separated by whitespace)_ with its respective parameters following the word.
+> - Each parameter must be entered following a flag, i.e. a name parameter will have the input `n/NAME`
+
+The following is a class diagram of the `Parser` and its relevant dependencies\
+![ParserClass](uml-diagrams/ParserClass.png)
+
+The following is a sequence diagram of the execution sequence of a single user input\
+![ParserSequence](uml-diagrams/ParserSequence.png)
+
+`Parser+parseCommand(input)` will return a `Command` object according to the command word in the user input.
+It returns an `InvalidCommand` class if the command word detected is invalid.
+
+For the parsing of valid command word inputs, `Parser+parseCommand(input)` passes the parameters of the user input into the respective command's parsing method, 
+then calls the method `Parser-getPatternMatcher(regex, input, paramFlags)`that returns a `Matcher` object, which is Java's regex pattern matcher.
+We use regular expressions to simplify the process of pattern matching for input parameters. Each command will have their respective regex strings for pattern matching
+
+> Note that `Parser-getPatternMatcher(regex, input, paramFlags)` will be called as long as the command word in the user input is valid,
+> even if the parameters are invalid. The `Matcher` object returned will try to match the input parameters to the given regex.
+> A `TrackerException` will be thrown if the input parameters are invalid.
+
+The input parameters will be passed into another method, `Parser-makeStringPattern(inputParam, paramFlags)` that returns a string in a pattern
+according to a preset format that contains all the necessary input parameters.
+Converting the parameter string to a pattern is done because **order does not matter** in the user's input parameters _(if there are multiple parameters)._
+Thus, it would be necessary to convert the user's input into a consistent format for pattern matching using regex.
+
+> Note: 
+> - All command regex are in the format:\
+>   `a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) ` with `a/`, `b/`, `c/` being the respective parameter flags
+>   and `/` as the flag character.
+>   - As of now, parameter flags must only contain 1 character right before the `/`
```

Comment:
Maybe it would be good to handle the error if the user enters more than 1 character, or edit the regex pattern string to specify that there must be only one character